src/lammpsio/snapshot.py

```diff
@@ -91,22 +91,77 @@ def from_hoomd_gsd(cls, frame):
         snap.charge = frame.particles.charge
         snap.mass = frame.particles.mass
 
-        snap.molecule = frame.particles.body + 1
+        if frame.particles.body is not None:
+            snap.molecule = frame.particles.body + 1
```

Comment:
Was this causing an issue? The default value of `body` should be -1 in HOOMD for something that doesn't have such a label, so it would become 0 meaning "not a molecule" to LAMMPS. I would have assumed that this if statement always evaluates True.

If the goal is not to write molecule info when it isn't present in the GSD file, we could be more sophisticated and check if `np.any(frame.particles.body != -1)`, then only do this molecule assigning and validation of the values, possibly also auto-remapping negative values of the body if they are all negative.

Reply:
Not sure what I was doing here. I will revert this change and we can address this later, if needed.

Reply:
I figured out why I was doing this. I get this error when using `to_hoomd_gsd` then going back with `from_hoomd_gsd`: 
``` 
snap.molecule = frame.particles.body + 1
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```

Reply:
Hmm, so which direction is causing the error, or what is the test case? I do see that it seems like the default value of any attribute in a `gsd.hoomd.Frame` is `None` if it has not been assigned. I'm not sure what happens when read from file though:
```py
>>> import gsd.hoomd
>>> frame = gsd.hoomd.Frame()
>>> print(frame.particles.body)
None
>>> print(frame.particles.position)
None
```

Reply:
The error pops up when using `from_hoomd_gsd` to read a GSD where body was never set. In lammpsio, it only gets set in `to_hoomd_gsd` here:
```
if self.has_molecule():
            frame.particles.body = self.molecule - 1
```

So when I was using lammpsio to go to GSD then back to LAMMPS data file, it causes as error.  I think we just need to check if it is not none before trying to set `snap.molecule`.

Reply:
Is this done by passing through a GSD file, though, or on the frame object directly?

I think checking for `None` is probably the right behavior, but I wanted to make sure I understood what was triggering it.

Reply:
I was passing the frame directly and it caused the error. I just checked writing to file and then passing a GSD file and that does not trigger the error. The body values default to -1.

Reply:
Got it, thanks!

I think that `from_hoomd_gsd` should work regardless of whether the `Frame` was created directly in Python or read from a GSD file, so I think this means that we actually need to do the `is not None` check on _every_ property before we assign it to the lammpsio `Snapshot`.

Similarly, on the `to_hoomd_gsd` side, we should only set the attributes of the GSD `Frame` if `has_X() is True` in the `Snapshot`. lammpsio does not have a default for attributes that are not set (just a default for how they are initialized), so I think we should just let GSD do its thing with those.